walkthroughs/howto-k8s-ingress-gateway/deploy.sh

```diff
@@ -0,0 +1,106 @@
+    echo "AWS_ACCOUNT_ID environment variable is not set."
+    exit 1
+fi
+
+if [ -z $AWS_DEFAULT_REGION ]; then
+    echo "AWS_DEFAULT_REGION environment variable is not set."
+    exit 1
+fi
+
+if [ -z $ENVOY_IMAGE ]; then
+    echo "ENVOY_IMAGE environment variable is not set to App Mesh Envoy, see https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html"
+    exit 1
+fi
+
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
+PROJECT_NAME="howto-k8s-ingress-gateway"
+APP_NAMESPACE=${PROJECT_NAME}
+MESH_NAME=${PROJECT_NAME}
+APPMESH_PREVIEW='"true"'
+
+ECR_IMAGE_PREFIX="${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/${PROJECT_NAME}"
+FRONT_APP_IMAGE="${ECR_IMAGE_PREFIX}/feapp"
+COLOR_APP_IMAGE="${ECR_IMAGE_PREFIX}/colorapp"
+MANIFEST_VERSION="${1:-v1beta2}"
+
+error() {
+    echo $1
+    exit 1
+}
+
+check_k8s_virtualgateway() {
+    #check CRD
+    crd=$(kubectl get crd virtualgateways.appmesh.k8s.aws -o json )
+    if [ -z "$crd" ]; then
+        error "$PROJECT_NAME requires virtualgateways.appmesh.k8s.aws CRD to support Ingress gateway. See https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/CHANGELOG.md"
+    else
+        echo "CRD check passed!"
+    fi
+}
+
+check_k8s_gatewayroutes() {
+    #check CRD
+    crd=$(kubectl get crd gatewayroutes.appmesh.k8s.aws -o json )
+    if [ -z "$crd" ]; then
+        error "$PROJECT_NAME requires gatewayroutes.appmesh.k8s.aws CRD to support Ingress gateway. See https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/CHANGELOG.md"
+    else
+        echo "CRD check passed!"
+    fi
+}
+
+check_appmesh_k8s() {
+    #check aws-app-mesh-controller version
+    if [ "$MANIFEST_VERSION" = "v1beta2" ]; then
+        currentver=$(kubectl get deployment -n appmesh-system appmesh-manager -o json | jq -r ".spec.template.spec.containers[].image" | cut -f2 -d ':'|tail -n1)
+        requiredver="v1.0.0"
+        check_k8s_virtualgateway
+        check_k8s_gatewayroutes
+    else
+        error "$PROJECT_NAME unexpected manifest version input: $MANIFEST_VERSION. Should be v1beta2 or v1beta1 based on the AppMesh controller version. See https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/CHANGELOG.md"
+    fi
+
+    if [ "$(printf '%s\n' "$requiredver" "$currentver" | sort -V | head -n1)" = "$requiredver" ]; then
+        echo "aws-app-mesh-controller check passed! $currentver >= $requiredver"
+    else
+        error "$PROJECT_NAME requires aws-app-mesh-controller version >=$requiredver but found $currentver. See https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/CHANGELOG.md"
+    fi
+}
+
+
+deploy_images() {
+    for app in colorapp feapp; do
+        aws ecr describe-repositories --repository-name $PROJECT_NAME/$app >/dev/null 2>&1 || aws ecr create-repository --repository-name $PROJECT_NAME/$app
+        docker build -t ${ECR_IMAGE_PREFIX}/${app} ${DIR}/${app}
+        $(aws ecr get-login --no-include-email)
```

Comment:
I think this wants to be:

    aws ecr get-login-password --region ${AWS_DEFAULT_REGION} | \                                                                                                        
        docker login --username AWS --password-stdin \ 
        ${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com

Not unique to this example though, looks like we need to fix this across all of the similar walkthroughs.

Reply:
+1 we do need to fix this across examples (which is outside the scope of these changes). You can support both awscli v1/2 [like this](https://github.com/aws/aws-app-mesh-examples/blob/master/walkthroughs/howto-timeout-policy/src/colorgateway/deploy.sh#L24-L28)